hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/BaseTokenHandler.java

```diff
@@ -259,28 +259,26 @@ private List<TokenRelation> createTokenRelsToAccount(
             // Link each of the new token IDs together in a doubly-linked list way by setting each
             // token relation's previous and next token IDs.
 
-            // Compute the previous and next token IDs. Unfortunately `TokenRelation` doesn't
-            // allow for null values, so a value of '0' will have to indicate a null pointer to
-            // the previous or next token (since no token number 0 can exist)
-            long prevTokenId = 0;
-            long nextTokenId = 0;
+            // Compute the previous and next token IDs.
+            TokenID prevTokenId = null;
+            TokenID nextTokenId = null;
             if (i - 1 >= 0) { // if there is a previous token
                 prevTokenId = Optional.ofNullable(tokens.get(i - 1))
```

Comment:
We have this pattern all over the place, and it's confusing to me why.
`Optional.ofNullable(tokens.get(i - 1)).map(Token::tokenId).orElse(null);`
is functionally identical to
`tokens.get(i - 1) != null ? tokens.get(i - 1) : null`
The latter avoids 3 object creations and is smaller and easier to follow.

Reply:
@jsync-swirlds I suppose "easier to follow" varies from person to person (I want a formula, dangit!). For those used to a functional paradigm, maybe the `Optional` way is easier to read. 

As for performance, I don't _think_ we have any empirical evidence that we're creating too many objects in modularized code paths

Reply:
Object creation is a fairly common concern because we use zGC in production, which is not currently generational (all collections are old-gen).  Oleg's performance testing doesn't use zGC because the GC pauses ruin the tests currently.
When we can easily avoid creating a stream, we should.  Streams should be used when we need them, not to avoid a ternary.

As for functional programming, I get it, but Java is not, and never will be, a functional language. Using the functional structures in Java is always going to be a bit confusing and jarring to the average Java engineer (and we need to consider that audience, because this system is _open source_, even if all of the contributions are internal at the moment).